### PR TITLE
Sg send timeout fix - followup

### PR DESCRIPTION
**Pull request by SreeramGarlapati**

### CODE REVIEW

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java [hunk 1]
```diff
@@ -251,35 +251,80 @@ private int getDataSerializedSize(Message amqpMessage)
...
 			{
-				applicationPropertiesSize += (value.toString().length() << 1);
+				applicationPropertiesSize += sizeof(value);
 			}
 		}
 		
 		return annotationsSize + applicationPropertiesSize + payloadSize;
 	}
+	
+	private static int sizeof(Object obj)
```
Comment (hmlam) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java]: >sizeof [](start = 20, length = 6)

make this an utility method.

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java [hunk 2]
```diff
@@ -521,8 +566,9 @@ public void onSendComplete(final Delivery delivery)
 			{
 				this.lastKnownLinkError = null;
 				this.retryPolicy.resetRetryCount(this.getClientId());
-				pendingSend.complete(null);
+
 				pendingSendWorkItem.getTimeoutTask().cancel(false);
+				pendingSend.complete(null);
```
Comment (hmlam) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java]: Can we just create a method like

public void completePendingSend(CompletableFuture<void> pendingSend, Exception exception) 

which do these patterns in one place?

**Comment by hmlam** [comment]
:clock1:

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java (hunk 2 above)
Comment (SreeramGarlapati) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java]: given, the timerTask and the actual CompletableFuture are really 2 different concepts - I would keep them as it is.

**Pull request merged**